"esmLexer: invalid syntax, require javascript/typescript" on valid file

I have a basic TypeScript file like this:

```
// file.ts
import { MyType, MyFunc } from "somemodule";
export const Foo = {} as const satisfies MyType;
export default MyFunc(Foo);
```
esm.sh (v135, running locally in docker) fails with this message:

```
2024/02/28 22:41:24 [error] build 'v135/../file.js': esmLexer: invalid syntax, require javascript/typescript
```

I'm wondering if the `satisfies` keyword is causing issues? I am not getting much else with loglevel=debug.
Strangely, files that depend on those exports are working fine, and I see `Foo` being inlined into them (which I don't want, but that's OK for now). I am expecting this file to still be addressable/exist.

OK it looks like this package (js_parser) is being invoked here: 
https://github.com/esm-dev/esm.sh/blob/main/server/utils.go#L205

It looks like that only validates JavaScript files using this:
https://github.com/ije/esbuild-internal/blob/master/js_parser/js_parser.go

Although, it seems to allow TypeScript files to be "found" and passed in. Writing a small test, I think this is the root cause:

```
--- FAIL: TestStrictMode (0.00s)
    --- FAIL: TestStrictMode/import{A,B}from"C";export_const_D={}_as_const_satisfies_A;export_default_B(D); (0.00s)
        js_parser_test.go:286: 
            +<stdin>: ERROR: Expected ";" but found "as"
             
```

The util should probably make use of the ts_parser if one of the ts extensions is selected:
https://github.com/ije/esbuild-internal/blob/master/js_parser/ts_parser.go

This was a confusing error because the message states require either javascript OR typescript, so it made me assume both were valid here. Upon inspection, I believe TS isn't properly supported.

--- a/server/utils.go
+++ b/server/utils.go
@@ -14,10 +14,10 @@ import (
 	"time"
 
 	"github.com/Masterminds/semver/v3"
+	"github.com/ije/esbuild-internal/config"
 	"github.com/ije/esbuild-internal/js_ast"
 	"github.com/ije/esbuild-internal/js_parser"
 	"github.com/ije/esbuild-internal/logger"
-	"github.com/ije/esbuild-internal/config"
 )
 
 const EOL = "\n"

--- a/server/utils.go
+++ b/server/utils.go
@@ -17,6 +17,7 @@ import (
 	"github.com/ije/esbuild-internal/js_ast"
 	"github.com/ije/esbuild-internal/js_parser"
 	"github.com/ije/esbuild-internal/logger"
+	"github.com/ije/esbuild-internal/config"
 )
 
 const EOL = "\n"
@@ -202,13 +203,18 @@ func validateJS(filename string) (isESM bool, namedExports []string, err error) 
 		return
 	}
 	log := logger.NewDeferLog(logger.DeferLogNoVerboseOrDebug, nil)
+	parserOpts := js_parser.OptionsFromConfig(&config.Options{
+		TS: config.TSOptions {
+			Parse: true,
+		},
+	})
 	ast, pass := js_parser.Parse(log, logger.Source{
 		Index:          0,
 		KeyPath:        logger.Path{Text: "<stdin>"},
 		PrettyPath:     "<stdin>",
 		Contents:       string(data),
 		IdentifierName: "stdin",
-	}, js_parser.Options{})
+	}, parserOpts)
 	if !pass {
 		err = errors.New("invalid syntax, require javascript/typescript")
 		return

